cargo test --all --all-targets --no-default-features --features serde fails since commit 94ca198b83f036e34650ff82f13b296c7969b9ba

After updating unicode-bidi in Debian to version 0.3.8 we noticed that "cargo test --all --all-targets --no-default-features --features serde" now fails with cargo test --all --all-targets --no-default-features --features serde

    error[E0599]: no function or associated item named `new` found for struct `BidiInfo` in the current scope 
     --> src/lib.rs:912:35
       |
    263 | pub struct BidiInfo<'text> {
          | -------------------------- function or associated item `new` not found for this
        ...
    912 |         let bidi_info = BidiInfo::new(text, None);
        |                                   ^^^ function or associated item not found in `BidiInfo<'_>`

I bisected the regression and found it comes from 94ca198b83f036e34650ff82f13b296c7969b9ba, this seems to be an actual regression, not merely a newly added test.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1088,7 +1088,7 @@ mod tests {
     }
 }
 
-#[cfg(all(feature = "serde", test))]
+#[cfg(all(feature = "serde", feature = "hardcoded-data", test))]
 mod serde_tests {
     use super::*;
     use serde_test::{assert_tokens, Token};

